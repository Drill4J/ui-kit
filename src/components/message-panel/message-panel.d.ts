import { Message } from './message-type';

interface Props {
  className?: string;
  message: Message;
  onClose(): void;
}
export declare const MessagePanel: {
  ({ className, message: { type, text }, onClose }: Props): JSX.Element;
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {
      className: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      message: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      onClose: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
    };
  };
};
export {};
