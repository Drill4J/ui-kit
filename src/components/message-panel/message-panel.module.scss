@import '../../style-guide/index';

.message-panel {
  height: 48px;
  width: 100%;
  position: absolute;
  right: 0;
  z-index: 100;

  &__content {
    position: absolute;
    display: flex;
    align-items: center;
    flex-grow: 1;
    height: 100%;
    width: 100%;
    color: map-get($monochrome, black);
    font-size: map-get($font-sizes, small);
    font-family: $semibold-font;
    line-height: 20px;

    &--type {
      &--success {
        background-color: map-get($green, light-tint);
      }

      &--error {
        background-color: map-get($red, ultralight-tint);
      }

      &--warning {
        background-color: map-get($orange, light-tint);
      }

      &--info {
        color: map-get($monochrome, default);
        background-color: map-get($monochrome, light-tint);
      }
    }
  }

  &__message-icon-wrapper {
    display: flex;
    align-items: center;
    margin-left: map-get($gutter-widths, extra-large);

    &--type {
      &--success {
        color: map-get($green, default);
      }

      &--error {
        color: map-get($red, default);
      }

      &--warning {
        color: map-get($orange, default);
      }

      &--info {
        color: map-get($monochrome, default);
      }
    }
  }

  &__message-text {
    display: flex;
    align-items: center;
    margin-left: map-get($gutter-widths, extra-small);
  }

  &__close-icon {
    margin-right: 25px;
    cursor: pointer;
  }
}
