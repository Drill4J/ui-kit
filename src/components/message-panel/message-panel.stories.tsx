import { storiesOf } from '@storybook/react';
import 'twin.macro';

import { MessagePanel } from './message-panel';
import { LinkButton } from '../forms';

storiesOf('MessagePanel', module)
  .add('SUCCESS', () => (
    <MessagePanel message={{ type: 'SUCCESS', text: 'Succes! Congratz!' }} onClose={() => {}} />
  ))
  .add('ERROR', () => (
    <MessagePanel message={{ type: 'ERROR', text: 'Error! Something went wrong!' }} onClose={() => {}} />
  ))
  .add('WARNING', () => (
    <MessagePanel
      message={{
        type: 'WARNING',
        text: (
          <div tw="flex items-center w-full">
            Warning! Something might go wrong!&nbsp;
            <LinkButton size="large">Undo</LinkButton>
          </div>
        ),
      }}
      onClose={() => {}}
    />
  ))
  .add('INFO', () => (
    <MessagePanel message={{ type: 'INFO', text: 'Info! This should be read!' }} onClose={() => {}} />
  ));
