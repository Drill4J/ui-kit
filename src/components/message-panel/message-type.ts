export interface Message {
  type: 'SUCCESS' | 'ERROR' | 'WARNING' | 'INFO';
  text: string;
}
