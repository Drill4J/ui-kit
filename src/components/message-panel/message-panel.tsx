import styled, { css } from 'styled-components';

import { Panel } from '../../layouts';
import { Icons } from '../icon';
import { Message } from './message-type';
import { COLORS, FONTS } from '../../theme';

interface Props {
  className?: string;
  message: Message;
  onClose(): void;
}

export const MessagePanel = ({ className, message: { type, text }, onClose }: Props) => (
  <Wrapper className={className}>
    <Content type={type}>
      <Panel align="space-between">
        <Panel>
          {getMessageIcon(type)}
          <MessageText>{text}</MessageText>
        </Panel>
        <CloseIcon onClick={onClose} />
      </Panel>
    </Content>
  </Wrapper>
);

const MessageText = styled.div`
  display: flex;
  align-items: center;
  margin-left: 8px;
`;

const CloseIcon = styled(Icons.Close)`
  margin-right: 25px;
  cursor: pointer;
`;

const Wrapper = styled.div`
  height: 48px;
  width: 100%;
  position: absolute;
  right: 0;
  z-index: 100;
`;

const Content = styled.div<Pick<Message, 'type'>>`
  position: absolute;
  display: flex;
  align-items: center;
  flex-grow: 1;
  height: 100%;
  width: 100%;
  color: ${COLORS.MONOCHROME.BLACK};
  font-size: 14px;
  font-family: ${FONTS.SEMI_BOLD};
  line-height: 20px;
  
  ${({ type }) => [
    type === 'SUCCESS' && css`background-color: ${COLORS.GREEN.LIGHT_TINT}`,
    type === 'ERROR' && css`background-color: ${COLORS.RED.ULTRALIGHT_TINT}`,
    type === 'INFO' && css`
      color: ${COLORS.MONOCHROME.DEFAULT};
      background-color: ${COLORS.MONOCHROME.LIGHT_TINT};
    `,
    type === 'WARNING' && css`background-color: ${COLORS.ORANGE.LIGHT_TINT}`,
  ]}
`;

const MessageIconWrapper = styled.div<Pick<Message, 'type'>>`
  display: flex;
  align-items: center;
  margin-left: 24px;
  
  ${({ type }) => [
    type === 'SUCCESS' && css`color: ${COLORS.GREEN.DEFAULT}`,
    type === 'ERROR' && css`color: ${COLORS.RED.DEFAULT}`,
    type === 'INFO' && css`color: ${COLORS.MONOCHROME.DEFAULT};`,
    type === 'WARNING' && css`color: ${COLORS.ORANGE.DEFAULT}`,
  ]}
`;

function getMessageIcon(type: 'SUCCESS' | 'ERROR' | 'WARNING' | 'INFO') {
  switch (type) {
    case 'ERROR': {
      return (
        <MessageIconWrapper type={type}>
          <Icons.Cancel height={16} width={16} />
        </MessageIconWrapper>
      );
    }
    case 'SUCCESS': {
      return (
        <MessageIconWrapper type={type}>
          <Icons.Checkbox height={16} width={16} />
        </MessageIconWrapper>
      );
    }
    case 'WARNING': {
      return (
        <MessageIconWrapper type={type}>
          <Icons.Warning height={16} width={16} />
        </MessageIconWrapper>
      );
    }
    default: {
      return (
        <MessageIconWrapper type={type}>
          <Icons.Info height={16} width={16} />
        </MessageIconWrapper>
      );
    }
  }
}
