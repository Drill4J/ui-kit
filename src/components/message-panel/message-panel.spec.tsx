import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MessagePanel } from './message-panel';
import { getByDataTest } from '../../../test-utils';

const fn = jest.fn();

describe('MessagePanel', () => {
  it('should call onClose function after click to close icon', () => {
    const { container } = render(<MessagePanel message={{ type: 'ERROR', text: 'TEXT' }} onClose={fn} />);
    fireEvent.click(getByDataTest(container, 'message-panel:close-icon'));
    expect(fn).toBeCalledTimes(1);
  });

  it('should match snapshot with type is ERROR', () => {
    const tree = renderer.create(
      <MessagePanel message={{ type: 'ERROR', text: 'TEXT' }} onClose={fn} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with type is SUCCESS', () => {
    const tree = renderer.create(
      <MessagePanel message={{ type: 'SUCCESS', text: 'TEXT' }} onClose={fn} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with type is WARNING', () => {
    const tree = renderer.create(
      <MessagePanel message={{ type: 'WARNING', text: 'TEXT' }} onClose={fn} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with type is INFO', () => {
    const tree = renderer.create(
      <MessagePanel message={{ type: 'INFO', text: 'TEXT' }} onClose={fn} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
