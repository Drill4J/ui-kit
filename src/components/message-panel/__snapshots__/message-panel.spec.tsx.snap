// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagePanel should match snapshot with the default text 1`] = `
<div
  className="sc-hKwCoD hGyoyU"
>
  <div
    className="sc-eCImvq jgjazv"
    type="SUCCESS"
  >
    <div
      className="panel panel--align--space-between"
      data-test=""
      onClick={[Function]}
    >
      <div
        className="panel"
        data-test=""
        onClick={[Function]}
      >
        <div
          className="sc-jRQAMF fULZPf"
          type="SUCCESS"
        >
          <svg
            className="sc-bdvvaa eCjinC"
            fillRule="evenodd"
            height={16}
            style={
              Object {
                "WebkitTransform": "rotate(0deg)",
              }
            }
            transform="rotate(0)"
            viewBox="0 0 20 20"
            width={16}
          >
            <path
              d="M 10 1 C 5.029 1 1 5.029 1 10 C 1 14.971 5.029 19 10 19 C 14.971 19 19 14.971 19 10 C 19 5.029 14.971 1 10 1 Z  M 10 0 C 15.523 0 20 4.477 20 10 C 20 15.523 15.523 20 10 20 C 4.477 20 0 15.523 0 10 C 0 4.477 4.477 0 10 0 Z  M 4.697 9.944 L 4.803 9.834 C 5.189 9.433 5.812 9.434 6.196 9.833 L 7.804 11.501 C 8.19 11.901 8.808 11.903 9.188 11.508 L 13.812 6.714 C 14.194 6.317 14.812 6.324 15.197 6.723 L 15.303 6.832 C 15.689 7.233 15.69 7.877 15.308 8.273 L 9.192 14.616 C 8.813 15.009 8.19 15.011 7.807 14.614 L 4.693 11.386 C 4.309 10.987 4.312 10.343 4.697 9.944 Z "
            />
          </svg>
        </div>
        <div
          className="sc-gsDJrp kbMxqn"
        >
          Succes! Congratz!
        </div>
      </div>
      <svg
        className="sc-bdvvaa eCjinC sc-dkPtyc ikLPDw"
        fillRule="evenodd"
        height={16}
        onClick={[Function]}
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
          }
        }
        transform="rotate(0)"
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M 8 7.242 L 15.205 0.073 C 15.462 -0.081 15.787 0.016 15.931 0.291 C 16.076 0.566 15.985 0.915 15.728 1.07 L 9.325 7.429 L 8.789 7.998 L 9.325 8.572 L 15.728 14.931 C 15.985 15.085 16.076 15.434 15.931 15.709 C 15.787 15.984 15.462 16.081 15.205 15.927 L 8 8.758 L 0.795 15.927 C 0.538 16.081 0.213 15.984 0.069 15.709 C -0.076 15.434 0.015 15.085 0.272 14.931 L 6.675 8.572 L 7.211 8.003 L 6.675 7.429 L 0.272 1.07 C 0.015 0.915 -0.076 0.566 0.069 0.291 C 0.213 0.016 0.538 -0.081 0.795 0.073 L 8 7.242 Z "
        />
      </svg>
    </div>
  </div>
</div>
`;
